src/components/AutoComplete/AutoComplete.component.tsx

```diff
@@ -135,6 +149,30 @@ export const AutoComplete = (props: AutoCompleteProps) => {
 		});
 	}, [searchQuery, getOptionsDelayed]);
 
+	/**
+	 * The function is called when an event is detected on the keyboard,
+	 * so you can browse through the list and select one.
+	 */
+	const onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+		if (event.key === 'Enter') {
+			const option = options[active];
+			setSearchValue(option.label ?? option.value);
+			setActive(0);
+			setOptions([]);
+			setOpen(false);
+    }
+		else if (event.key === 'ArrowUp') {
+      if (active === 0) return;
+			setActive(prev => prev - 1);
+			listRef.current.scrollBy(0, -Constants.ITEM_HEIGHT);
+    }
+		else if (event.key === 'ArrowDown') {
+      if (active === options.length - 1) return;
+      setActive(prev => prev + 1);
+			listRef.current.scrollBy(0, Constants.ITEM_HEIGHT);
+    }
```

Comment:
```suggestion
		if (event.key === 'Enter') {
			const option = options[active];
			setSearchValue(option.label ?? option.value);
			setActive(0);
			setOptions([]);
			setOpen(false);
			return;
    }
    
	if (event.key === 'ArrowUp') {
            if (active === 0) return;
			setActive(prev => prev - 1);
			listRef.current.scrollBy(0, -Constants.ITEM_HEIGHT);
			return;
    }
    
    if (event.key === 'ArrowDown') {
            if (active === options.length - 1) return;
            setActive(prev => prev + 1);
			listRef.current.scrollBy(0, Constants.ITEM_HEIGHT);
			return;
    }
```

we should prevent `else if` by using `return` if possible this makes code more readable and prevents dependency flow